operator/v1/types_machineconfiguration.go

```diff
@@ -56,3 +65,94 @@ type MachineConfigurationList struct {
+// such as the resource type and the API Group of the resource. It also provides granular control via the selection field.
+type MachineManager struct {
+	// resource is the machine management resource's type.
+	//
+	// Valid value(s): machinesets
+	//
+	// * machinesets means that the machine manager will only register resources of the type MachineSet, which may belong to MachineAPI or ClusterAPI.
+	//
+	// +kubebuilder:validation:Required
+	Resource MachineManagerMachineSetsResourceType `json:"resource"`
+	// apiGroup is name of the APIGroup that the machine management resource belongs to.
+	//
+	// Valid value(s):  machine.openshift.io
+	//
+	// * machine.openshift.io means that the machine manager will only register resources that belong to MachineAPI APIGroup.
+	//
+	// +kubebuilder:validation:Required
+	APIGroup MachineManagerMachineSetsAPIGroupType `json:"apiGroup"`
+	// selection allows granular control of the machine management resources that will be registered for boot image updates.
+	//
+	// +kubebuilder:validation:Required
+	Selection MachineManagerSelector `json:"selection"`
+}
+
+// +kubebuilder:validation:XValidation:rule="has(self.mode) && self.mode == 'Partial' ?  has(self.partial) : !has(self.partial)",message="Partial is required when type is partial, and forbidden otherwise"
+// +union
+type MachineManagerSelector struct {
+	// mode determines how machine managers will be selected for updates.
+	//
+	// Valid value(s): All, Partial.
+	//
+	// * All means that every resource matched by the machine manager will be updated.
+	//
+	// * Partial requires a specified selector and allows customisation of which resources matched by the machine manager will be updated.
+	// +unionDiscriminator
+	// +kubebuilder:validation:Required
+	Mode MachineManagerSelectorMode `json:"mode"`
+
+	// partial provides a label selector that can be used to match machine management resources.
+	// Only permitted when mode is set to "Partial".
+	// +optional
+	Partial *metav1.LabelSelector `json:"partial,omitempty"`
+}
+
+// MachineManagerSelectorMode is a string enum used in the MachineManagerSelector union discriminator.
+// +kubebuilder:validation:Enum:="All";"Partial"
+type MachineManagerSelectorMode string
+
+const (
+	// All represents a configuration mode that registers all resources specified by the parent MachineManager for boot image updates.
+	All MachineManagerSelectorMode = "All"
+
+	// Partial represents a configuration mode that will register resources specified by the parent MachineManager only
+	// if they match with the label selector.
+	Partial MachineManagerSelectorMode = "Partial"
+)
+
+// MachineManagerManagedResourceType is a string enum used in the MachineManager type to describe the resource
+// type to be registered.
+// +kubebuilder:validation:Enum:="machinesets"
+type MachineManagerMachineSetsResourceType string
+
+const (
+	// machinesets represent the MachineSet resource type, which manage a group of machines.
```

Comment:
Nit, comment on this one should be capitalised
```suggestion
	// MachineSets represent the MachineSet resource type, which manage a group of machines.
```